migrations/2020_03_03_060845_combine_mentions.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Builder;
+
+
+return [
+    'up' => function (Builder $schema) {
+        // Create new table
+        $schema->create('post_mentions', function (Blueprint $table) {
+            $table->integer('post_id')->unsigned();
+            $table->integer('mentions_post_id')->unsigned()->nullable();
+            $table->integer('mentions_user_id')->unsigned()->nullable();
+            $table->primary(['post_id']);
+        });
+
+        // Establish foreignkey constraints
+        $schema->table('post_mentions', function (Blueprint $table) {
+            $table->foreign('post_id')->references('id')->on('posts')->onDelete('cascade');
+            $table->foreign('mentions_user_id')->references('id')->on('users')->onDelete('cascade');
+            $table->foreign('mentions_post_id')->references('id')->on('posts')->onDelete('cascade');
+        });
+
+        // Move stuff from post mentions table
+        $postMentions = $schema->getConnection()->table('post_mentions_post')->cursor();
+
+        foreach($postMentions as $postMention) {
+            $schema
+                ->getConnection()->table('post_mentions')
+                ->insert([
+                    'post_id' => $postMention->post_id,
+                    'mentions_post_id' => $postMention->mentions_post_id,
+                    'mentions_user_id' => null
+                ]);
+        }
```

Comment:
You should be able to achieve this with an [`INSERT... SELECT` query](https://dev.mysql.com/doc/refman/8.0/en/insert-select.html), I think.

That should be more performant and safer.

Not sure whether the query builder supports this type of query. If not, raw queries can be used as well (though we have to be careful to apply table prefixes if we do so).

Reply:
The query builder does not. However, as an alternative, we could do a separate batch and an insert. Or we could chunk it. Raw SQL is doable, and I've done so here: https://github.com/askvortsov1/flarum-categories/blob/master/src/Console/RecalculateTagStats.php, but it's chaotic at the least.